### PR TITLE
Refractor avatar unit tests to use react testing library

### LINKED ISSUE #81: [ui-components] refactor Avatar tests with RTL
- dependent on #49 
- refactor tests to implement RTL instead of Enzyme

### PR DESCRIPTION
**Pull request by tharun208**
# Refractor avatar unit tests to use react testing library

 - [x] Add `react testing library` library to package.json
 - [x] refractor `Avatar` component unit tests from to use `react testing library` and not  `enzyme`.

#### Checklist
- [x] I have written or updated the test for the changes I made
- [ ] I have updated the README of the package I'm working in to reflect my changes
- [ ] I have added or updated Storybook if appropriate for my changes

Fixes #81

### CODE REVIEW

**Comment by tharun208** [comment]
@asubiotto, can I get a review on this.

**Comment by asubiotto** [comment]
Hi @tharun208, I'm not the best person to review this. cc @dhartunian

**Comment by dhartunian** [comment]
Hi @tharun208 thanks for making these changes and submitting the PRs.

Your PRs are in the review pipeline so give us a few days to get to them. We're in the middle of wrapping up a release cycle for the DB so our response time on reviews will be a bit slow. If we haven't gotten to it by next Monday, please feel free to ping me again :) Appreciate your patience.

File: packages/ui-components/src/Avatar/Avatar.test.tsx [hunk 1]
```diff
@@ -1,75 +1,91 @@
 import React from "react";
-import { shallow } from "enzyme";
+import { render, fireEvent, screen } from "@testing-library/react";
 
 import Avatar, { AvatarIntent, AvatarSize } from "./Avatar";
```
Comment (nathanstilwell) [inline packages/ui-components/src/Avatar/Avatar.test.tsx:4]: I don't want to pollute the component with a data attribute, but I agree that having one on the component is convenient for testing. So maybe we could wrap the component from the lib to add a data attribute in this file for testing purposes. I'm imagining something like,

```jsx
import UiAvatar, { AvatarIntent, AvatarSize, AvatarProps } from "./Avatar";

const Avatar: React.FC<AvatarProps> = ({ children, ...props }) => (
  <UiAvatar data-testid="avatar" {...props}>
    {children}
  </UiAvatar>
);
```

**Review by nathanstilwell — COMMENTED** [review]
The tests look great, but I would rather not add a data attribute to the component. I left a comment for a possible solution that would minimize the refactoring.

File: packages/ui-components/src/Avatar/Avatar.test.tsx (hunk 1 above)
Comment (tharun208) [reply packages/ui-components/src/Avatar/Avatar.test.tsx]: updated the tests @nathanstilwell

**Pull request merged**